### ISSUE TITLE
Deleting Vertices - Wrong Points Removed on Trash

### ISSUE THREAD

**Issue by mikeomeara1** [mapbox/mapbox-gl-draw#897]
**mapbox-gl-js version**: 1.0
**mapbox-gl-draw version**: 1.0.9

### Steps to Trigger Behavior

 1. Go to:  https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-draw/
 2. Draw a polygon with more than ~20 points
 3. Select more than 10 vertices starting at the 10th point or later
 4. Click the "Trash" button
 4. At least one (in cases with polygons that have many vertices - many) of the selected points still remain while some un-selected points are removed.

### Expected Behavior
All **selected** points are removed when in `direct_select` mode and the `trash` event is executed.

### Actual Behavior
Some selected points will still remain. Some un-selected points are removed.

### Example 1 - Large number of points
![image](https://user-images.githubusercontent.com/8380449/59642403-b3e74980-9119-11e9-9e7e-ffc9b4d617b2.png)
Yields the following after `trash`:
![image](https://user-images.githubusercontent.com/8380449/59643278-eba3c080-911c-11e9-8135-7cf63b613d1e.png)

### Example 2 - Small number of points
![image](https://user-images.githubusercontent.com/8380449/59643105-586a8b00-911c-11e9-8b23-707be19d5ee2.png)
After Trash:
![image](https://user-images.githubusercontent.com/8380449/59643243-c020d600-911c-11e9-8c75-17d45ac23bdd.png)

### Remediation
I'm not certain, but I believe this has something to do with the `coord_path` assigned to every vertex. When the `trash` event fires on `direct_select` it does:
```javascript
state.selectedCoordPaths.sort().reverse().forEach(id => state.feature.removeCoordinate(id));
```
I believe that the `sort()` function doesn't take into account that `coord_path` is a string and when the values get greater than `"0.10"` the sort function puts the vertices in the wrong order, thus removes the wrong ones.

**Comment by mikeomeara1** [comment]
After troubleshooting, it appears if you change the following line in `direct_select.js` (line 168) from:

```javascript
state.selectedCoordPaths.sort().reverse().forEach(id => state.feature.removeCoordinate(id));
```

To:

```javascript
 state.selectedCoordPaths.map(p => {return {coord_path: p, sort_val: parseInt(p.split('.')[p.split(".").length -1], 10)}})
    .sort((a, b) => a.sort_val - b.sort_val)
    .map(p => p.coord_path)
    .reverse()
    .forEach(id => state.feature.removeCoordinate(id));
```

It appears to resolve the issue (as ugly as it may be). After I've had a chance to test a bit further, we'll deploy this fix to our fork for community review.

If anyone has optimizations or other ideas...they'd be most welcome.

**Comment by kkaefer** [comment]
@mikeomeara1 thanks for the detailed analysis, it was very helpful in creating a fix in #943.
